main.rb

```diff
@@ -0,0 +1,71 @@
+require 'open3'
+
+def get_env_variable(key)
+  return (ENV[key] == nil || ENV[key] == "") ? nil : ENV[key]
+end
+ 
+def env_has_key(key)
+	value = get_env_variable(key)
+	return value unless value.nil? || value.empty?
+
+	abort("Input #{key} is missing.")
+end
+
+def run_command(command)
+	puts "@@[command] Executing: #{command}"
+	stdout_str, stderr_str, status = Open3.capture3(command)
+
+	if status.success?
+		puts stdout_str unless stdout_str.empty?
+		puts stderr_str unless stderr_str.empty?
+		return stderr_str.empty? ? stdout_str : stderr_str
+	else
+		puts "@@[error] Command failed:\n#{stderr_str}"
+		raise stderr_str
+	end
+end
+
+def abort_with0(message)
+  puts "@@[error] #{message}"
+  exit 0
+end
+
+def abort_with1(message)
+	puts "@@[error] #{message}"
+	exit 1
+end
+
+def get_available_java_versions()
+	all_java_versions = [8, 11, 17, 21, 23]
+	available_java_versions = []
+
+	all_java_versions.each do |java_version|
+		available_java_versions << java_version unless !get_env_variable("JAVA_HOME_#{java_version}_X64")
+	end
+
+	available_java_versions.join(', ')
+end
+
+sdkman_dir = env_has_key('SDKMAN_DIR')
```

Comment:
I could not see this env var in the reserved env docs. Is this reserved env? If it is yes, should we add to doc? I am not completely sure.

Reply:
@boztopuz  Yes, this is actually an Environment Variable for the server. We use it to change the JDK version on the system. It was also used in the doc [here](https://docs.appcircle.io/workflows/common-workflow-steps/custom-script#changing-system-java-version)

Is there a reason we added this to reserved?

Reply:
Upon installation, SDKMAN creates an environment variable `$SDKMAN_DIR`, which points to the installation directory.

Also, you can use that environment variable before installation to customize the SDKMAN installation path. [ref](https://sdkman.io/install/#install-to-a-custom-location)

It can be added to the reserved environment variables, but we cannot change its naming since it's also used by SDKMAN internally.